xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less

```diff
@@ -131,4 +131,81 @@ img {
   -moz-hyphens: auto;
   -webkit-hyphens: auto;
   hyphens: auto;
+}
+
+// User preferences ===========================================================
+body.content {
+  /* We apply various user preference related styles to the content of the page.
+   We use classes that are added to the body itself in htmlheader.vm */
+  // XWiki font size preference
+  &.preference-font-size-regular {
+    font-size: 140%;
+  }
+
+  &.preference-font-size-large {
+    font-size: 160%;
+  }
+
+  &.preference-font-size-larger {
+    font-size: 180%;
+  }
+
+  &.preference-font-size-largest {
+    font-size: 200%;
```

Comment:
I'm wondering if 200% isn't breaking the layout too much currently.

Reply:
I updated it to `1.4*@font-size-base`. 
Note that this 200% wasn't actually 200% of the default size but 200/140, which was only a 42% increase.

With a `@font-size-base` of 20px (above default), zoom level 100%, on a 2k display I get:
![21492-largestNew](https://github.com/xwiki/xwiki-platform/assets/28761965/f2882aef-9436-4589-810d-41e5986c80ae)
I used the Firefox device size emulation dev tool to get this display in the same conditions for a 1920*1080 screen:
![21492-largest19201080](https://github.com/xwiki/xwiki-platform/assets/28761965/930e80f8-6499-4f67-8a9f-bf0d93b5ab06)


IMO this looks okay, a bit off in some places but not breaking the layout.
Note that for most users, the text will be smaller than that, this is a worst case scenario.

Reply:
This was done with the changes pushed in https://github.com/xwiki/xwiki-platform/pull/2694/commits/09af3b4516658f4e8b3356a9c64fce626a0d607b

Reply:
To me the whole spacing just looks completely wrong. Also, the fact that not all text is scaled and in particularly the smallest text is not scaled at all makes this feature kind of useless to me.

I'm wondering if we shouldn't just have a vote to drop this font size increase feature in favor of pointing users to the browser's zoom functionality until we have proper font scaling support.

Reply:
I figured out why... The values with `0.9*@font-size-base` would not be modified by this...
E.g. the `last modified` is the same size whatever the preference.

We'd need to have the value of font-size-base change based on the preferences, but AFAIK this is not something we want to do because we would need to recompile all the LESS of the codebase for each user... which was the exact thing we tried to avoid by using a class on the body node...


So :+1: to start a vote, and implement this font size feature back when we have a codebase consistent enough to make it possible.

Reply:
I think what might be worth investigating is if we could replace the less variable `@font-size-base` by a CSS variable without changing the whole logic too much. This might allow easily changing the base font size with full effects everywhere without needing any less compilation.

Reply:
So, I just checked it out, and LESS is not really made with that in mind...
First, I came up with escaping issues, the `--` used to define a CSS variable were interpreted as the start of `-->` and failed the LESS compilator, so I had to escape all of those characters:
```
#elseif($property == 'font-size-base')
  ## Special case, we rely on a CSS variable to control the value so that we can update it with more ease.
  :root {\-\-$property: $value;}
  @$property: var(\-\-$property);
```
in variablesInit.vm
and
```
&.preference-font-size-regular {
  \-\-font-size-base: calc(1*var(\-\-font-size-base));
}
```
in general.less

However, this workaround was not enough, because after that compilation failed with the message:


> Caused by: com.github.sommeri.less4j.Less4jException: Could not compile less. 270 error(s) occurred:
> ERROR less/bootstrap/variables.less 67:34 Wrong argument type to function 'floor', expected number saw LIST_EXPRESSION.
>  66: //** Computed "line-height" (`font-size` * `line-height`) for use with `margin`, `padding`, etc.
>  67: @line-height-computed:    floor((@font-size-base * @line-height-base)); // ~20px
>  68: 

Bootstrap uses the value of the @font-size-base to calculate the value of @line-height-computed, but uses LESS syntax to do so. This syntax is not compatible with CSS variables :disappointed:

Reply:
> Bootstrap uses the value of the @font-size-base to calculate the value of @line-height-computed, but uses LESS syntax to do so. This syntax is not compatible with CSS variables 😞

My idea would have been to also replace `@line-height-computed` by a variable that is initialized as CSS `calc()` expression.